src/Controls/src/Core/Platform/Android/BottomNavigationViewUtils.cs

```diff
@@ -104,6 +104,8 @@ internal static void UpdateEnabled(bool tabEnabled, IMenuItem menuItem)
 			if (showMore)
 			{
 				var moreString = context.Resources.GetText(Resource.String.overflow_tab_title);
+				if (menu.Size() == maxBottomItems)
+					menu.RemoveItem(menu.GetItem(menu.Size() - 1).ItemId);
```

Comment:
I feel like in this else check we need to see if we've hit the "more" item and then remove iit if we do
https://github.com/dotnet/maui/pull/22589/files#diff-1787c6c4f33d04c0dd8bceddf563be45c136658b564e5a6be28161f37b79adc5R94-R99

like, if the user has 6 items it'll be
0,1,2,3,more

if they go down to 5 items then I think the more tab won't swap in the correct 5th item

And then instead of "Removing" this item. Can we retrieve the item using the MoreTabId to see if it was still added?

Reply:
@PureWeen you're right

<video src="https://github.com/dotnet/maui/assets/42434498/0a0d85d5-ec51-4947-8d7b-745ac8b3e0b9" width="300px"/>